databaseManager.class.php

```diff
@@ -0,0 +1,157 @@
+        abstract protected function deleteData($table, $conditions);
+
+        /**
+         * Constructor
+         */
+        protected function __construct($dbDriver, $host, $port, $username, $password, $dbName, $unixSocket, $charset)
+        {
+            $this->dbDriver = $dbDriver;
+            $this->host = $host;
+            $this->port = $port;
+            $this->username = $username;
+            $this->password = $password;
+            $this->dbName = $dbName;
+            $this->unixSocket = $unixSocket;
+            $this->charset = $charset;
+        }
+
+        /**
+         * @return bool
+         */
+        public function isConnected()
+        {
+            return ($this->connection instanceof PDO);
+        }
+
+        /**
+         * Kill connection
+         */
+        public function closeConnection()
+        {
+            $this->connection = null;
+        }
+
+        /**
+         * Start transaction
+         */
+        public function startTransaction()
+        {
+            $this->connection->beginTransaction();
+        }
+
+        /**
+         * Commit the changes
+         */
+        public function commitTransaction()
+        {
+            $this->connection->commit();
+        }
+
+        /**
+         * Rollback the changes
+         */
+        public function rollbackTransaction()
+        {
+            $this->connection->rollBack();
+        }
+
+        /**
+         * Store cache file in cache folder
+         * @param string name of cache file
+         * @param array data from select query
+         */
+        public function storeCacheFile($sqlCacheName, $data)
+        {
+            /* Write data to file */
+            $filew = fopen("cache/" . $sqlCacheName, 'w');
+            fwrite($filew, print_r($data, true));
+            fclose($filew);
+        }
+
+        /**
+         * Checks if cache file exists and is modified less than one hour
+         * @param string cache file
+         * @param int total minutes before expiration
+         * @return bool
+         */
+        public function isCached($cacheFile, $cacheTimeSeconds)
+        {
+            return (file_exists($cacheFile) && (filemtime($cacheFile) > (time() - ($cacheTimeSeconds))));
+        }
+
+        /**
+         * Execute functionality for query caching
+         * @param string select query
+         * @return array
+         */
+        public function initCacheArray($sql)
+        {
+            /*Generate an MD5 hash from the SQL query above.*/
+            $cacheArray['sqlCacheName'] = md5($sql) . ".cache";
+  
+            /* The name of our cache folder. */
+            $cacheArray['cache'] = 'cache';
+                       
+            /* Full path to cache file. */
+            $cacheArray['cacheFile'] = $cacheArray['cache'] . "/" . $cacheArray['sqlCacheName'];
+            
+            /* Cache time in seconds. 60 * 60 = one hour. */
+            $cacheArray['cacheTimeSeconds'] = (60 * 60);
```

Comment:
@lumakos Why are you hardcoding the TTL of the cached data?

Reply:
It's my first time working with query caching , I read about it and memcache. I am not sure for my solution , i just try to implement something about query caching.